### ISSUE TITLE
CA1801 emitted for deserialization constructor when implementing ISerializable.

### ISSUE THREAD

**Issue by alphaleonis** [dotnet/roslyn-analyzers#3039]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.7 (Latest)

#### Diagnostic ID

Example: [1801](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1801?view=vs-2019)

#### Repro steps

Consider the following class:
```
   [Serializable]
   public class Class1 : ISerializable
   {
      private int _i;

      protected Class1(SerializationInfo info, StreamingContext context)
      {
         _i = (int)info.GetValue("i", typeof(int));
      }

      public void GetObjectData(SerializationInfo info, StreamingContext context)
      {
         if (info == null)
            throw new ArgumentNullException(nameof(info), $"{nameof(info)} is null.");
       
         info.AddValue("i", _i);
      }
   }
```

This warns about `context` being an unused parameter in the ctor, and suggests removing it. But the `ISerializable` interface implies that this ctor exists with exactly that signature according to the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.iserializable?view=netframework-4.8).  

#### Expected behavior

No warning about unused parameters suggesting to remove them should be emitted.

#### Actual behavior

CA1801 is emitted for the `context` parameter of the ctor.

**Comment by lhj42** [comment]
Experiencing the same

**Comment by Evangelink** [comment]
Makes sense to me but we should probably wait for #3223 to be merged first.
